src/migrator/sync.ts

```diff
@@ -0,0 +1,177 @@
+	defaultMigrationCategory,
+	checkModelAlreadyExists,
+	setExecutedMigrations,
+	getExecutedMigrations,
+	lockMigrations,
+	MigrationCategories,
+	RunnableMigrations,
+} from './utils';
+import type { Tx } from '../database-layer/db';
+import type { Config, Model } from '../config-loader/config-loader';
+
+import * as _ from 'lodash';
+import * as sbvrUtils from '../sbvr-api/sbvr-utils';
+
+type ApiRootModel = Model & { apiRoot: string };
+
+export const postRun = async (tx: Tx, model: ApiRootModel): Promise<void> => {
+	const { initSql } = model;
+	if (initSql == null) {
+		return;
+	}
+
+	const modelName = model.apiRoot;
+
+	const exists = await checkModelAlreadyExists(tx, modelName);
+	if (!exists) {
+		(sbvrUtils.api.migrations?.logger.info ?? console.info)(
+			'First time executing, running init script',
+		);
+
+		await lockMigrations(tx, modelName, async () => {
+			try {
+				await tx.executeSql(initSql);
+			} catch (err) {
+				(sbvrUtils.api.migrations?.logger.error ?? console.error)(
+					`postRun locked sql execution error ${err} `,
+				);
+				throw new MigrationError(err);
+			}
+		});
+	}
+};
+
+export const run = async (tx: Tx, model: ApiRootModel): Promise<void> => {
+	const { migrations } = model;
+	if (
+		migrations == null ||
+		_.isEmpty(migrations) ||
+		migrations[defaultMigrationCategory] === undefined
+	) {
+		return;
+	}
+
+	let runMigrations: RunnableMigrations = {};
+	for (const [key, value] of Object.entries(migrations)) {
+		if (key === defaultMigrationCategory && value instanceof Object) {
+			runMigrations = { ...runMigrations, ...value };
+		} else if (!(key in MigrationCategories)) {
+			runMigrations = { ...runMigrations, ...{ [key]: value } };
+		}
+	}
+	return $run(tx, model, runMigrations);
+};
+
+const $run = async (
+	tx: Tx,
+	model: ApiRootModel,
+	migrations: RunnableMigrations,
+): Promise<void> => {
+	const modelName = model.apiRoot;
+
+	// migrations only run if the model has been executed before,
+	// to make changes that can't be automatically applied
+	const exists = await checkModelAlreadyExists(tx, modelName);
+	if (!exists) {
+		(sbvrUtils.api.migrations?.logger.info ?? console.info)(
+			'First time model has executed, skipping migrations',
+		);
+
+		return await setExecutedMigrations(tx, modelName, Object.keys(migrations));
+	}
+	await lockMigrations(tx, modelName, async () => {
+		try {
+			const executedMigrations = await getExecutedMigrations(tx, modelName);
+			const pendingMigrations = filterAndSortPendingMigrations(
+				migrations,
+				executedMigrations,
+			);
+			if (pendingMigrations.length === 0) {
+				return;
+			}
+
+			const newlyExecutedMigrations = await executeMigrations(
+				tx,
+				pendingMigrations,
+			);
+			await setExecutedMigrations(tx, modelName, [
+				...executedMigrations,
+				...newlyExecutedMigrations,
+			]);
+		} catch (err) {
+			(sbvrUtils.api.migrations?.logger.error ?? console.error)(
+				`$run executedMigrations error ${err}`,
```

Comment:
Can you change this message so that it's clearer to a user what actually failed and therefore where they should be looking for the issue in their code